### PR TITLE
Made PHPCR initializer independent, creating all route_basepaths

### PR DESCRIPTION
**Pull request by pamil**
Related PR Sylius/Sylius#3668.

### CODE REVIEW

**Comment by pamil** [comment]
Failed only on PHP7, may be connected with enabled XDebug on Travis.

File: composer.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
         "matthiasnoback/symfony-dependency-injection-test": "~0.6",
         "matthiasnoback/symfony-config-test": "0.*",
         "phpunit/php-code-coverage": "@stable",
+        "phpunit/phpunit": "^4.0",
```
Comment (WouterJ) [inline composer.json]: please do not depend on phpunit

File: Resources/config/initializer-phpcr.xml [hunk 2]
```diff
@@ -4,13 +4,15 @@
...
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
+    <parameters>
+        <parameter key="cmf_routing.initializer.class">Doctrine\Bundle\PHPCRBundle\Initializer\GenericInitializer</parameter>
+    </parameters>
+
     <services>
 
-        <service id="cmf_routing.initializer" class="Doctrine\Bundle\PHPCRBundle\Initializer\GenericInitializer">
+        <service id="cmf_routing.initializer" class="%cmf_routing.initializer.class%">
```
Comment (WouterJ) [inline Resources/config/initializer-phpcr.xml]: class parameters are no longer recommended, please revert that change.

**Comment by dbu** [comment]
thanks!
i restarted the builds that failed. as master is not failing, i think it should work. if it still does not, we can try restarting the build for master to see if with the final release of php 7, there was some change that makes us break.

**Comment by pamil** [comment]
I disabled XDebug and it passes :)

**Comment by pjedrzejewski** [comment]
:+1:

**Comment by WouterJ** [comment]
So there is now a bC break in this PR: While route basepaths were not initialized by default before, they are now. I think we the RoutingBundle should keep using the old configuration and not initialize the basepaths unless configured.

**Comment by dbu** [comment]
agreed. lets say that initializer defaults to `auto` and auto means if
sonata is active, we do the initializer, otherwise not. and if its
true/false we just do or not do the initializer.

**Comment by dbu** [comment]
@pamil can you adjust the PR to use `auto` for the initializer?

**Comment by pamil** [comment]
@dbu Sure, thanks for reminding me this PR, I almost forgot about it. I'm on my way! :)

**Comment by pamil** [comment]
@dbu should be up and running now.

**Comment by pamil** [comment]
Only StyleCI is failing and it's (mostly) not related to my changes.

**Comment by dantleech** [comment]
@pamil you should be able to rebase: try (on your branch):

```
$ git fetch origin
$ git rebase origin/master
```

If it encounters any conflicts you will need to fix them and then:

```
$ git add /fixed/files
$ git rebase --continue
```

If you have any difficulties let us know

**Comment by pamil** [comment]
@dantleech thanks, I must have missed something, now the rebase went all easy :) Style CI is passing, tests should be passing.

**Comment by dbu** [comment]
great, this looks good now!
can you please squash the commits into 1 commit plus 1 for the travis config change? (git rebase -i... then reorder the commits and squash the last two into the first one)

**Comment by pamil** [comment]
@dbu done :)

**Comment by dbu** [comment]
thanks. one last thing: can you please do a pull request on the symfony-cmf-docs repository to update the configuration references of the routing bundle? i think its enough to mention the feature there. but  if you want you can look over the routing documentation and see if there is a place where it would make sense to mention this feature. please use the `versionadded` functionality (this feature will be in version 1.4)

**Comment by dbu** [comment]
thanks a lot. created https://github.com/symfony-cmf/symfony-cmf-docs/issues/724 for the documentation.

**Comment by pjedrzejewski** [comment]
:+1:

**Pull request merged**